feat: add local subcommand to compare two chart folders

bo0tzz opened the pull request:
This adds a new `local` subcommand that diffs two local chart directories. My intended usecase is mainly for development and CI/CD, to see what effects a change has.

This code was written entirely by an LLM; if that is a dealbreaker, feel free to just close this PR outright.

If there is appetite for it, I'd like to next build on top of this with a `git` subcommand that diffs two git references against eachother (by checking them out to a temporary folder and then running `local` against them).

bo0tzz committed:
feat: add local subcommand to compare two chart folders

bo0tzz committed:
fix: format

yxxhero commented:
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/databus23/helm-diff/pull/866#pullrequestreview-3321907397)

yxxhero committed:
fix: address PR review comments for local subcommand

- Lowercase error messages to follow Go conventions
- Extract stdin reading into prepareStdinValues() to avoid
  double-read when rendering both charts
- Add HELM_BIN env var validation with "helm" default fallback

Signed-off-by: yxxhero <aiopsclub@163.com>

yxxhero committed:
fix: pass []byte directly to manifest.Parse instead of converting to string

Signed-off-by: yxxhero <aiopsclub@163.com>

yxxhero committed:
fix: check error return values for errcheck linter

Signed-off-by: yxxhero <aiopsclub@163.com>

yxxhero committed:
test: add integration tests for local subcommand

Signed-off-by: yxxhero <aiopsclub@163.com>

yxxhero committed:
fix: use errors.As instead of type assertion for errorlint

Signed-off-by: yxxhero <aiopsclub@163.com>

yxxhero commented:
see: https://github.com/databus23/helm-diff/pull/982
